src/queue/consumer.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@nestjs/common'
+import { SqsMessageHandler } from '@ssut/nestjs-sqs'
+import { Message } from '@aws-sdk/client-sqs'
+import { QueueMessage } from './queue.types'
+
+@Injectable()
+export class ConsumerService {
+  constructor() {}
+  @SqsMessageHandler(process.env.SQS_QUEUE || '', false)
+  async handleMessage(message: Message) {
+    const shouldRequeue = await handleMessageAndMaybeRequeue(message)
+    // Return a rejected promise if requeue is needed without throwing an error
+    if (shouldRequeue) {
+      return Promise.reject('Requeuing message without stopping the process')
+    }
+    return true // Return true to delete the message from the queue
+  }
+}
+
+// Function to process message and decide if requeue is necessary
+async function handleMessageAndMaybeRequeue(
+  message: Message,
+): Promise<boolean> {
+  try {
+    await handleMessage(message) // Your main processing logic
+    return false // No requeue needed
+  } catch (error) {
+    console.error('Message processing failed, will requeue:', error)
+    return true // Indicate that we should requeue
+  }
+}
+
+async function handleMessage(message: Message) {
```

Comment:
This is confusing.. why are there 2 `handleMessage` methods in this file? 🤔 
<img width="729" alt="Screenshot 2024-12-12 at 1 29 40 PM" src="https://github.com/user-attachments/assets/53c6f061-71f9-4e9c-982d-45294e4910f5" />